AmplifyTests/CategoryTests/API/APICategoryConfigurationTests.swift

```diff
@@ -250,13 +256,64 @@ class APICategoryConfigurationTests: XCTestCase {
             plugins: ["MockAPICategoryPlugin": true]
         )
 
-        try Amplify.API.configure(using: categoryConfig)
+        guard let api = Amplify.API as? CategoryConfigurable & Resettable else {
+            XCTFail("API is not CategoryConfigurable and Resettable")
+            return
+        }
+
+        try api.configure(using: categoryConfig)
 
         let semaphore = DispatchSemaphore(value: 0)
-        Amplify.API.reset { semaphore.signal() }
+        api.reset { semaphore.signal() }
         semaphore.wait()
 
-        XCTAssertNoThrow(try Amplify.API.configure(using: categoryConfig))
+        XCTAssertNoThrow(try api.configure(using: categoryConfig))
+    }
+
+    func testIsConfiguredIsFalseBeforeConfig() {
+        guard let category = Amplify.API as? AmplifyAPICategory else {
+            XCTFail("Could not cast Amplify.API as AmplifyAPICategory")
+            return
+        }
+        XCTAssertFalse(category.isConfigured)
+    }
+
+    func testIsConfiguredIsTrueAfterConfig() throws {
+        let plugin = MockAPICategoryPlugin()
+        try Amplify.add(plugin: plugin)
+        let categoryConfig = APICategoryConfiguration(
+            plugins: ["MockAPICategoryPlugin": true]
+        )
+
+        let amplifyConfig = AmplifyConfiguration(api: categoryConfig)
+        try Amplify.configure(amplifyConfig)
+
+        guard let category = Amplify.API as? AmplifyAPICategory else {
+            XCTFail("Could not cast Amplify.API as AmplifyAPICategory")
+            return
+        }
+
+        XCTAssertTrue(category.isConfigured)
```

Comment:
LGTM, on a related note; I'll do some refactoring with how dataStore actually syncs to the cloud via API plugin to remove the dependency on checking Model.isSyncable when adding selection set. there should be enough information to construct the request with the sync-able fields. This is needed for API plugin's Model based APIs because they are not sync-able by default. Then `category.isConfigured` check is only used for datastore to check if it should sync to the cloud. Datastore constructing the graphQL request should have enough information to add sync-able fields and shouldn't need to know whether or not the category is configured.

1. Developer using API by itself should never provision a sync-able API
2. Developer using DataStore should only provision a sync-able API